docs/source/1.0/spec/core/type-refinement-traits.rst

```diff
@@ -138,6 +138,64 @@ in Java).
         }
 
 
+.. smithy-trait:: smithy.api#input
+.. _input-trait:
+
+---------------
+``input`` trait
+---------------
+
+Summary
+    Specializes a structure for use only as the input of a single operation.
+Trait selector
+    ``structure``
+Value type
+    Annotation trait.
+Conflicts with
+    :ref:`output-trait`, :ref:`error-trait`
+
+Structure shapes marked with the ``@input`` trait MUST adhere to the
+following constraints:
+
+1. They can only be referenced in the model as an operation's input.
+2. They cannot be used as the input of more than one operation.
+3. They SHOULD have a shape name that starts with the name of the
+   operation that targets it (if any). For example, the input shape of the
+   ``GetSprocket`` operation SHOULD be named ``GetSprocketInput``.
```

Comment:
Yeah I can reword this a bit. We are inherently nudging people towards using Input/Output suffixes because they will be the default suffix used by the inline operation input and output syntax, but it's configurable.